src/System.Drawing.Common/tests/Helpers.cs

```diff
@@ -27,21 +27,24 @@ public static bool GetGdiplusIsAvailable()
             }
             else
             {
-                IntPtr nativeLib;
+                IntPtr libgdiplusHandle;
                 if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
                 {
-                    nativeLib = dlopen("libgdiplus.dylib", RTLD_NOW);
+                    libgdiplusHandle = dlopen("libgdiplus.dylib", RTLD_NOW);
                 }
                 else
                 {
-                    nativeLib = dlopen("libgdiplus.so", RTLD_NOW);
-                    if (nativeLib == IntPtr.Zero)
+                    libgdiplusHandle = dlopen("libgdiplus.so", RTLD_NOW);
+                    if (libgdiplusHandle == IntPtr.Zero)
                     {
-                        nativeLib = dlopen("libgdiplus.so.0", RTLD_NOW);
+                        libgdiplusHandle = dlopen("libgdiplus.so.0", RTLD_NOW);
                     }
                 }
 
-                return nativeLib != IntPtr.Zero;
+                int dlcloseResult = dlclose(libgdiplusHandle);
```

Comment:
Not necessarily correct: please change this to

```
if(libgdiplusHandle != IntPtr.Zero)
{
  dlclose(libgdiplusHandle);
}
```

Reply:
Makes sense, changed to close conditionally.